app/models/miq_expression.rb

```diff
@@ -995,7 +995,7 @@ def self.get_col_info(field, options = {})
     if col
       result[:data_type] = col_type(model, col)
       result[:format_sub_type] = MiqReport::FORMAT_DEFAULTS_AND_OVERRIDES[:sub_types_by_column][col.to_sym] || result[:data_type]
-      result[:virtual_column] = true if model.virtual_attribute?(col.to_s)
+      result[:virtual_column] = true if !model.attribute_in_sql?(col.to_s)
```

Comment:
This looks good in theory, but it seems unfortunate there are virtual columns for which this will not be set, so this could have potential for surprise. Not sure the best way to mitigate this. Maybe comment?